articles/constraint-satisfaction-minizinc-gecode/index.md

```diff
@@ -0,0 +1,360 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/constraint-satisfaction-minizinc-gecode/
+title: Constraint-Satisfaction Problems with MiniZinc + Gecode
+description: Constraint-Satisfaction Problems (CSPs) represent problems in terms of variables, their possible values, and constraints on those values. In this tutorial, we will use MiniZinc, a programming language built for CSPs, and Gecode, a CSP solver, to model and solve a simple map-coloring problem.
+author: nicholas-kross
+date: 2021-01-04T00:00:00-20:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/constraint-satisfaction-minizinc-gecode/hero.jpg
+    alt: united states map
+---
+Constraint-Satisfaction Problems (CSPs) represent problems as variables, their possible values, and constraints on those values. Many problems, from graph theory to type inference to puzzle-solving, are converted into CSPs. In this tutorial, we will use MiniZinc, a programming language built for CSPs, and Gecode, a CSP solver, to model and solve a simple graph-coloring problem.
```

Comment:
Many of the sentences in the article can be split into shorter sentences for easier reading.

Reply:
Alright I think I fixed those!